### PR TITLE
Static selectors version made primary

### LINKED ISSUE #20: Add static selectors amount assigner

### PR DESCRIPTION
**Pull request by nkaskov**
Closes #20

### CODE REVIEW

**Review by makxenov — APPROVED** [review]

**Comment by NickVolynkin** [comment]
If I understand correctly, submodule updates are required for the new code to work. In such cases, it's best to update them in the same commit with code changes. This way each of your commits has "green" code. If you remove the intermediate merge commit and other dependency updates, there should be just one commit with all the changes.

File: bin/circuit_transpiler/src/main.cpp [hunk 1]
```diff
@@ -0,0 +1,495 @@
...
+          "ar_params.constant_columns,\n"
+       << "                    0\n"
+       << "                );\n"
+       << "        }\n"
+       << "\n"
+       << "        uint256 t = 0;\n"
+       << "        assembly {\n"
+       << "            let modulus := mload(gate_params)\n"
+       << "            let theta_acc := 1\n"
+       << "            function get_eval_i_by_rotation_idx(idx, rot_idx, ptr) -> result {"
```
Comment (NickVolynkin) [inline bin/circuit_transpiler/src/main.cpp:116]: this line doesn't have `\n` in the end

File: bin/circuit_transpiler/src/main.cpp [hunk 2]
```diff
@@ -0,0 +1,495 @@
...
+}
+
+template<typename FRIParamsType, typename TableDescriptionType, typename ColumnsRotationsType,
+         typename ArithmetizationParams>
+void print_params(FRIParamsType &fri_params, TableDescriptionType table_description,
+                  ColumnsRotationsType &columns_rotations, std::string output_folder) {
+    std::ofstream out;
+
+    out.open(output_folder + "/circuit_params.json");
+    out << "{\"arithmetization_params\":[" << ArithmetizationParams::witness_columns << ","
```
Comment (NickVolynkin) [inline bin/circuit_transpiler/src/main.cpp:151]: Looks like you're building JSON with primitive tools. Isn't there any JSON library for converting data structures?

File: bin/circuit_transpiler/src/main.cpp [hunk 3]
```diff
@@ -0,0 +1,495 @@
...
+
+    for (size_t i = 0; i < witness.size(); i++) {    // witnesses.size() == ArithmetizationParams.WitnessColumns
+        ColumnType column;
+        typename BlueprintFieldType::integral_type num;
+        for (size_t j = 0; j < rows_amount; j++) {
+            istr >> num;
+            column.push_back(typename BlueprintFieldType::value_type(num));
+        }
+        witness[i] = column;
+    }
```
Comment (NickVolynkin) [inline bin/circuit_transpiler/src/main.cpp:218]: This code is repeated four times with minimal changes. Could it be a function?

The comment is always about `witnesses.size`, but there's no `witnesses` variable. Instead, there's `witness` (note the difference), `public_input` and others.

File: bin/circuit_transpiler/src/main.cpp [hunk 4]
```diff
@@ -0,0 +1,495 @@
...
+    return step_list;
+}
+
+template<typename FRIScheme, typename FieldType>
+typename FRIScheme::params_type create_fri_params(std::size_t degree_log, const int max_step = 1) {
+    typename FRIScheme::params_type params;
+    nil::crypto3::math::polynomial<typename FieldType::value_type> q = {0, 0, 1};
+
+    constexpr std::size_t expand_factor = 0;
+    std::size_t r = degree_log - 1;
```
Comment (NickVolynkin) [inline bin/circuit_transpiler/src/main.cpp:281]: `r` is not really obvious, could you use a more explicit name?

File: bin/circuit_transpiler/src/main.cpp [hunk 5]
```diff
@@ -0,0 +1,495 @@
...
+    out.open(output_file);
+    print_hex_byteblob(out, cv.cbegin(), cv.cend(), false);
+}
+
+int main(int argc, char *argv[]) {
+    if (argc < 2) {
+        std::cout << "There are two commands:" << std::endl
+                  << "gen_test_proof prepares gate argument, placeholder params and sample proof for testing"
+                  << std::endl
+                  << "gen_gate_argument prepares gate argument and some placeholder params" << std::endl;
```
Comment (NickVolynkin) [inline bin/circuit_transpiler/src/main.cpp:328]: Is this a help string for something like `transpiler --help`? Let's work on it a little more.

First, what are "some" placeholder params? Does the first command prepare a full set of params?

Second, what is the usage scenario? When will user see this message?

**Pull request merged**